src/mono/browser/runtime/diagnostics/client-commands.ts

```diff
@@ -0,0 +1,442 @@
+
+import { SessionId } from "./common";
+import { runtimeHelpers } from "./globals";
+
+// ADVR_V1\0
+export const advert1 = [65, 68, 86, 82, 95, 86, 49, 0,];
+// DOTNET_IPC_V1\0
+export const dotnet_IPC_V1 = [68, 79, 84, 78, 69, 84, 95, 73, 80, 67, 95, 86, 49, 0];
+
+// this file contains the IPC commands that are sent by client (like dotnet-trace) to the diagnostic server (like Mono VM in the browser)
+// just formatting bytes, no sessions management here
+
+
+export function advertise () {
+    // xxxxxxxx-xxxx-4xxx-xxxx-xxxxxxxxxxxx
+    const uuid = new Uint8Array(16);
+    crypto.getRandomValues(uuid);
+    uuid[7] = (uuid[7] & 0xf) | 0x40;// version 4
+
+    const pid = runtimeHelpers.mono_wasm_process_current_pid();
+
+    return Uint8Array.from([
+        ...advert1,
+        ...uuid,
+        ...serializeUint64([0, pid]),
+        0, 0// future
+    ]);
+}
+
+export function commandStopTracing (sessionID:SessionId) {
+    return Uint8Array.from([
+        ...serializeHeader(CommandSetId.EventPipe, EventPipeCommandId.StopTracing, computeMessageByteLength(8)),
+        ...serializeUint64(sessionID),
+    ]);
+}
+
+export function commandResumeRuntime () {
+    return Uint8Array.from([
+        ...serializeHeader(CommandSetId.Process, ProcessCommandId.ResumeRuntime, computeMessageByteLength(0)),
+    ]);
+}
+
+export function commandProcessInfo3 () {
+    return Uint8Array.from([
+        ...serializeHeader(CommandSetId.Process, ProcessCommandId.ProcessInfo3, computeMessageByteLength(0)),
+    ]);
+}
+
+
+export function commandGcHeapDump (extraProviders:DiagnosticCommandProviderV2[]) {
+    return commandCollectTracing2({
+        circularBufferMB: 256,
+        format: 1,
+        requestRundown: true,
+        providers: [
+            {
+                keywords: [
+                    0x0000_0000,
+                    Keywords.GCHeapSnapshot, // 0x1980001
+                    // GC_HEAP_DUMP_VTABLE_CLASS_REF_KEYWORD 0x8000000
+                    // GC_FINALIZATION_KEYWORD               0x1000000
+                    // GC_HEAP_COLLECT_KEYWORD               0x0800000
+                    // GC_KEYWORD                            0x0000001
+                ],
+                logLevel: 5,
+                provider_name: "Microsoft-Windows-DotNETRuntime",
+                arguments: null
+            },
+            ...extraProviders,
+        ]
+    });
+}
+
+export function commandCounters (intervalSec:number, extraProviders:DiagnosticCommandProviderV2[]) {
+    return commandCollectTracing2({
+        circularBufferMB: 256,
+        format: 1,
+        requestRundown: false,
+        providers: [
+            {
+                keywords: [0, Keywords.GCHandle],
+                logLevel: 4,
+                provider_name: "System.Diagnostics.Metrics",
+                arguments: `SessionId=SHARED;Metrics=System.Runtime;RefreshInterval=${intervalSec};MaxTimeSeries=1000;MaxHistograms=10;ClientId=c98f989b-369c-41af-bc8e-7ab261fba16c`
```

Comment:
Does `ClientId=c98f989b-369c-41af-bc8e-7ab261fba16c` ever need to change? (I'm not 100% sure what ClientId means in this context)